swap_meet/vendor.py

```diff
@@ -1,2 +1,132 @@
-class Vendor:
-    pass
\ No newline at end of file
+from swap_meet.item import Item
+class Vendor(Item):
+    def __init__(self, inventory = None):
+        super().__init__(category='')
+        if inventory is None:
+            inventory = []
+        self.inventory = inventory
+        
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        for check_item in self.inventory:
+            if check_item == item:
+                self.inventory.remove(item)
+                return item
+        return False
+    # Wave 2 
+    def get_by_category(self, category):
+        result = []
+        for each_item in self.inventory:
+            if each_item.category == category:
+                result.append(each_item)
+        return result
+    
+    # Wave 3
+    def swap_items(self, friend, my_item, their_item):
+        my_swap = False
+        friend_swap = False
+        for i in self.inventory:
+            if i == my_item:
+                my_swap = True
+        for fi in friend.inventory:
+            if fi == their_item:
+                friend_swap = True
+        
+        if my_swap == True and friend_swap == True:  
+            self.remove(my_item)
+            friend.add(my_item)
+            friend.remove(their_item)
+            self.add(their_item)     
+            return self.inventory 
```

Comment:
The instructions asked for you to return `True`, not the inventory of self. The tests still pass because we do `assert result`, which checks to see if the result is truth-y, which a non-empty list is, but this isn't what this function should do.